tutorials/developing_with_ign-cmake.md

```diff
@@ -0,0 +1,197 @@
+\page developingwithcmake Developing with Ignition CMake 
+
+# Developing with Ignition CMake
+
+This tutorial documents various tips and strategies for developing with Ignition CMake.
+
+## Helpful CMake flags
+
+There are several flags that control the results of the CMake tool.
+Some of these flags are built into CMake, where some are Ignition CMake specific.
+
+### Setting the build type
+
+The `CMAKE_BUILD_TYPE` variable controls the type of binary output from the build stage.
+This will have an impact on the flags passed to the compiler and linker.
+
+The available options are:
+
+  * `RelWithDebInfo` (default): Mostly optimized build, but with debug symbols.
+  * `Debug`: Debug build without optimizations, with debug symbols enabled.
+  * `Release`: Fully optimized build, no debug symbols.
+  * `MinSizeRel`: Fully optimized build, minimal binary size 
+  * `Coverage`: Build with additional information required for code coverage computation
+  * `Profile`: Use flags that are helpful with the `gprof` profiling tool
```

Comment:
Can we add an example of how to set one of these flags, like you did with `CMAKE_EXPORT_COMPILE_COMMANDS` in the [Creating a compilation database](https://github.com/ignitionrobotics/ign-cmake/blob/217011749bfb00a56bd3d46e0a7d1c9aedf47eeb/tutorials/developing_with_ign-cmake.md#creating-a-compilation-database) section? I'd think that including something like `-DCMAKE_BUILD_TYPE=Debug` would be sufficient.